conans/util/dates.py

```diff
@@ -38,15 +37,25 @@ def timestamp_to_str(timestamp):
     return datetime.datetime.utcfromtimestamp(int(timestamp)).strftime('%Y-%m-%d %H:%M:%S UTC')
 
 
-def timedelta_from_text(interval):
-    match = re.search(r"(\d+)([smhdw])", interval)
+def timelimit(expression):
+    """ convert an expression like "2d" (2 days) or "3h" (3 hours) to a timestamp in the past
+    with respect to current time
+    """
+    time_value = expression[:-1]
     try:
-        value, unit = match.group(1), match.group(2)
-        name = {'s': 'seconds',
-                'm': 'minutes',
-                'h': 'hours',
-                'd': 'days',
-                'w': 'weeks'}[unit]
-        return datetime.timedelta(**{name: float(value)})
-    except Exception:
-        raise ConanException("Incorrect time interval definition: %s" % interval)
+        time_value = int(time_value)
+    except TypeError:
+        raise ConanException(f"Time value '{time_value}' must be an integer")
+    time_units = expression[-1]
+    units = {"w": 7*24*60*60,
+             "d": 24*60*60,
+             "h": 60*60,
+             "m": 60,
+             "s": 1}
```

Comment:
Don't we want to accept years here as well?

Reply:
Too much? I don't think users will be wanting to remove artifacts only those artifacts that are +1 year old, running with the same cache for more than 1 year without cleaning sounds highly improbable. They can always do ``--lru=365d`` too

Reply:
Ok, I commented that because my first use of the feature was doing a `--lru=1y` and I was surprised it did not exist.

Reply:
Added month M and year 'y'